### PR TITLE
feature(frontend): adding open search data store form

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the open search form to the settings page.

## Changes

- Includes validation logic
- OpenSearch form
- Fixes with the wrong commit from the prev merge

## Fixes

- [[In-App Config] Setup Wizard - Data Store Step](https://github.com/kubeshop/tracetest/issues/1627)

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/dd61369bdb4d4d8eba714bab5171bee2

### CODE REVIEW

File: web/src/components/Settings/DataStorePlugin/forms/OpenSearch/OpenSearch.tsx [hunk 1]
```diff
@@ -1,5 +1,31 @@
+import {Form, Input} from 'antd';
+import * as S from '../../DataStorePluginForm.styled';
+import AddressesList from './AddressesList';
+
 const OpenSearch = () => {
-  return <div>OpenSearch</div>;
+  return (
+    <S.FormContainer>
+      <S.FormColumn>
+        <Form.Item label="Username" name="username">
```
Comment (jorgeepc) [inline web/src/components/Settings/DataStorePlugin/forms/OpenSearch/OpenSearch.tsx:9]: The names from this form does not match the `TDraftConfig` structure. This is going to create a different object structure than the expected in the config Services.

File: web/src/components/Settings/DataStorePlugin/forms/OpenSearch/AddressesList.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  add(): void;
+  remove(name: number): void;
+}
+
+const AddressesList = ({fields, add, remove}: IProps) => {
+  return (
+    <>
+      <S.ItemListContainer>
+        {fields.map(({key, name, ...field}, index) => (
+          <S.ListItem>
```
Comment (jorgeepc) [inline web/src/components/Settings/DataStorePlugin/forms/OpenSearch/AddressesList.tsx:17]: We need a `key` prop here.

**Review by jorgeepc — COMMENTED** [review]
Looks good so far. We are not using the `<SetupConfigProvider />` in the code so validation and submit is not working right now. Also, there is a problem with the form types and we are creating a different structure for the draft values.

File: web/src/providers/SetupConfig/SetupConfig.provider.tsx [hunk 3]
```diff
@@ -39,10 +39,10 @@ const SetupConfigProvider = ({children}: IProps) => {
 
   const onSaveConfig = useCallback(
     async (draft: TDraftConfig) => {
-      const configRequest = await SetupConfigService.getRequest(draft);
-      const config = await updateConfig(configRequest).unwrap();
+      console.log('@@saving', draft);
 
-      console.log('@@config', config);
+      // const configRequest = await SetupConfigService.getRequest(draft);
+      // const config = await updateConfig(configRequest).unwrap();
```
Comment (jorgeepc) [inline web/src/providers/SetupConfig/SetupConfig.provider.tsx:45]: Is this intentional?

**Review by jorgeepc — APPROVED** [review]
💣

File: web/src/providers/SetupConfig/SetupConfig.provider.tsx (hunk 3 above)
Comment (xoscar) [reply web/src/providers/SetupConfig/SetupConfig.provider.tsx:45]: Yep

**Pull request merged**